.github/workflows/build-snap.yml

```diff
@@ -0,0 +1,38 @@
+name: Snap Builder
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+  # Allow manual trigger
+  workflow_dispatch:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        platform: 
+          - arm64
+    steps:
+        
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Setub QEMU
+        uses: docker/setup-qemu-action@v2
+        with:
+          platforms: ${{ matrix.platform }}
+
+      - name: Build snap for arm64
+        uses: diddlesnaps/snapcraft-multiarch-action@v1
+        id: build
+        with:
+          architecture: ${{ matrix.platform }}
+      
+      - name: Upload artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: matter-pi-gpio-commander_${{ github.run_number}}_arm64.snap
```

Comment:
```suggestion
          name: matter-pi-gpio-commander_${{ github.run_number}}_${{ matrix.platform }}.snap
```
For consistency.

Reply:
I'm not sure if this will work.

The `${{ steps.build.outputs.snap }}` variable provides us with the `path` for the artifact directly from the previous step. 

My understanding is:
- GitHub Actions takes the value from `path` and assigns the `name` for uploading the artifact. 

Or it's not like that?

Reply:
Are we talking about the same thing?

The snap name comes from the value of the name attribute. I'm simply suggesting to take the architecture string from the matrix, just like the qemu and build steps.

Reply:
@farshidtz Sorry, I misunderstood. You're right.